src/herder/HerderImpl.cpp

```diff
@@ -726,19 +734,55 @@ HerderImpl::sendSCPStateToPeer(uint32 ledgerSeq, Peer::pointer peer)
     ZoneScoped;
     bool log = true;
     auto maxSlots = Herder::LEDGER_VALIDITY_BRACKET;
+
+    auto sendSlot = [=](SCPEnvelope const& e, bool log) {
```

Comment:
I missed this during my earlier review, but I think this can be problematic on shutdown: Peer is kept alive via shared_ptr, so we still call sendMessage. The issue is that half of the application is destroyed at that point, so there are no guarantees that things will fail gracefully. The reason this was not an issue in master is because the lambda was executed synchronously, whereas with this change we're scheduling a callback for later execution (possibly after shutdown is called).  To make this safe:
* Use weak_ptr, no-op if peer is destroyed by the time lambda is called
* No need to capture everything by value, you might accidentally capture other strong references as the code evolves, which is risky when dealing with async tasks. Safer to explicitly only capture the weak_ptr.